psmc_check.py

```diff
@@ -211,6 +213,14 @@ def main(opt):
                 plots_validation=plots_validation)
 
 
+def calc_off_nom_rolls(states):
+    off_nom_rolls = []
+    for state in states:
+        att = [state[x] for x in ['q1', 'q2', 'q3', 'q4']]
+        time = (state['tstart'] + state['tstop']) / 2
```

Comment:
Mid-time is correct.  The spacecraft attitude does not change during the observation, only the sun position.  So mid-time is the most reasonable (simple) way of getting off-nominal roll for the state.  At the end of the day small variations don't matter at all for thermal modeling.